### PR TITLE
Improvements to the design and defaults

### PR DESCRIPTION
**Pull request by abhi18av**
Hey there :wave:

This Pull-request address important enhancements focusing on better usability and easier user interaction.

# Main changes:

- Promote QC to main workflow
- Deprecate sra support (point to fetchngs)
- Move default_params
- Rename batch -> normal analysis
- Add the custom container build mechanism
- Standardize the deployment of Conda
- Remove necessity of user dealing with GATK-jar when using docker or conda.

### CODE REVIEW

**Comment by Mxrcon** [comment]
@abhi18av on my latest commits I tried to standardize the overall deployment of containers and conda environment as well. On `conda` step I added a step to register the GATK jar and check MTBseq installation. 
I think that the overall goals of this Pull request will be addressed by this, so I'm Kindly requesting for a review.

I published the image here: https://github.com/orgs/mtb-bioinformatics/packages/container/package/mtbseq-nf, It's published as a `mtb-bioinformatics` package instead of `mtbseq-nf` package image, as It may be complex to explain that this image is required to run mtbseq-nf but isn't the mtbseq-nf tool containerized.

**Comment by abhi18av** [comment]
Task linked: [CU-2zaw8mw Finalize the improvements to design and defaults](https://app.clickup.com/t/2zaw8mw)

File: conda_envs/setup_conda_envs.sh [hunk 1]
```diff
@@ -1,9 +1,32 @@
...
+cp GenomeAnalysisTK-3.8-0-ge9d806836/GenomeAnalysisTK.jar .
+
+echo "Creating mtbseq-nf-env"
+mamba env create -p mtbseq-nf-env --file mtbseq-nf-env.yml
+
+echo "Registering GATK Jar"
+
+eval "$(conda shell.bash hook)"
+conda activate "./mtbseq-nf-env"
+gatk-register GenomeAnalysisTK.jar
```
Comment (abhi18av) [inline conda_envs/setup_conda_envs.sh:25]:
```suggestion
# TODO: Good candidate for a clean approach in a refactor.
eval "$(conda shell.bash hook)"
conda activate "./mtbseq-nf-env"
gatk-register GenomeAnalysisTK.jar 
```

**Comment by abhi18av** [comment]
Merging this PR after local testing

**Pull request merged**